numpy/distutils/system_info.py

```diff
@@ -498,17 +505,46 @@ def calc_libraries_info(self):
                 dict_append(info, **i)
             else:
                 log.info('Library %s was not found. Ignoring' % (lib))
+            i = self.check_libs(r_dirs, [lib])
+            if i is not None:
+                # Swap library keywords found to runtime_library_dirs
+                # the libraries are insisting on the user having defined
+                # them using the library_dirs, and not necessarily by
+                # runtime_library_dirs
+                del i['libraries']
+                i['runtime_library_dirs'] = i.pop('library_dirs')
+                dict_append(info, **i)
```

Comment:
Yes, but that would also add the libraries key, which should be provided by the library key and library_dirs, not intrinsically by runtime_library_dirs. I did this so that library_dirs has precedence for actual libraries, and runtime_library_dirs only works for those directories listed.
So it is a "hack" to not re-create a check_libs routine only for the runtime libs.
The way I did it it will only allow runtime_library_dirs to be added iff the libraries also exists in those paths (this is what check_libs does).

If your above code would be used, all libraries would be added twice to the list. Also we need to transfer library_dirs to runtime_library_dirs, hence the pop to the other key.

###### site1.cfg
```
[lapack]
libraries = lapack
library_dirs = /opt/lapack
```
Linking command: `... -L/opt/lapack -llapack`.  
Here `LD_LIBRARY_PATH` is needed if shared lapack is shared.
###### site2.cfg
```
[lapack]
libraries = lapack
library_dirs = /opt/lapack
runtime_library_dirs = /opt/lapack
```
Linking command: `... -L/opt/lapack -Wl,-rpath=/opt/lapack -llapack`.  
Here the runtime library directory will be set, `LD_LIBRARY_PATH` not needed any more.
###### site3.cfg
```
[lapack]
libraries = lapack
runtime_library_dirs = /opt/lapack
```
Linking command: `... -Wl,-rpath=/opt/lapack -llapack`.